README.md

```diff
@@ -24,3 +24,54 @@ end
 3. Commit your changes (`git commit -am 'Add some feature'`)
 4. Push to the branch (`git push origin my-new-feature`)
 5. Create new Pull Request
+
+## Testing
+
+### Installing the Rubies
+
+For setting up a dev environment with all the Rubies on it, it is convenient to
+use the [rake-compiler-dev-box](https://github.com/tjschuck/rake-compiler-dev-box).
+
+
+### Caveats: Encodings
+
+Some tests fail if the external/locale/filesystem encoding is ISO-8859-1, which is the default on
+the [rake-compiler-dev-box](https://github.com/tjschuck/rake-compiler-dev-box).
+
+The tests on Travis run with certain [ENV variables our suite depends on](http://docs.travis-ci.com/user/ci-environment/):
+
+> LANG=en_US.UTF-8
+> LC_ALL=en_US.UTF-8
+> JRUBY_OPTS="--server -Dcext.enabled=false -Xcompile.invokedynamic=false"
+
+Ruby 1.9+ on *nix uses `LANG`, `LC_ALL` among other variables to set the Encoding.default_external.
+The settings above will set it to UTF-8.  Thus, before running tests, you'll want to ensure you or your
+login profile have run
+
+```bash
+export LANG=en_US.UTF-8
+export LC_ALL=en_US.UTF-8
+```
+
+Ruby 1.9+ on Windows, the Encoding.default_external is likely to be `ibm437`.  You can get the current `codepage`
+from the command prompt by running `chcp` and change it to utf8 via `chcp 65001`. See
+https://github.com/rspec/rspec-support/pull/151#discussion_r22991439
+http://stackoverflow.com/questions/1259084/what-encoding-code-page-is-cmd-exe-using and
+https://github.com/ruby/ruby/blob/9fd7afefd04134c98abe594154a527c6cfe2123b/ext/win32ole/win32ole.c#L540
+for more information and https://github.com/rspec/rspec-support/pull/151#discussion_r22991439 for
+background discussion.
+
+The above platform-specific methods are necessary when running our tests on Ruby 1.8.7.  Otherwise,
+we could run our tests commands with `ruby -E UTF-8:UTF-8 -S $spec_command`, which fails on Ruby 1.8.7.
+
+
+Before continuing, we should not that in https://github.com/rspec/rspec-support/blob/v3.2.1/script/functions.sh#L9
+we append `-X-C` to the JRUBY_OPTS.
+
+Available JRUBY options can be reviewed by running `jruby --properties`.
+
+### Running tests
+
+The easiest way to run the tests locally just like on Travis is to use the [wwtd (What Would Travis Do) gem](https://github.com/grosser/wwtd).
```

Comment:
> I'm surprised this hasn't been an issue for you.

I think it might have been an issue once.  But 99% of the time, when things fail on travis it's not because of the travis environment -- it's because of a bug in the spec, implementation, or an issue with a specific interpreter, all of which can be demonstrated locally with a simple `rspec` command.